src/telliot_feeds/reporters/tellorflex.py

```diff
@@ -217,46 +210,58 @@ async def get_num_reports_by_id(self, query_id: bytes) -> Tuple[int, ResponseSta
         count, read_status = await self.oracle.read(func_name="getNewValueCountbyQueryId", _queryId=query_id)
         return count, read_status
 
-    async def fetch_datafeed(self) -> Optional[DataFeed[Any]]:
+    async def rewards(self) -> int:
         tip = 0
-
-        if self.datafeed is not None:
-            # Skip fetching datafeed & checking profitability
-            if self.expected_profit == "YOLO":
-                return self.datafeed
-
-            single_tip, status = await self.autopay.get_current_tip(self.datafeed.query.query_id)
-            if not status.ok:
-                logger.warning("Unable to fetch single tip")
-                return None
-
+        datafeed: DataFeed[Any] = self.datafeed  # type: ignore
+        # Skip fetching datafeed & checking profitability
+        if self.expected_profit == "YOLO":
+            return tip
+
+        single_tip, status = await self.autopay.get_current_tip(datafeed.query.query_id)
+        if not status.ok:
+            logger.warning("Unable to fetch single tip")
+        else:
             tip += single_tip
 
-            feed_tip = await get_feed_tip(self.datafeed.query.query_id, self.autopay)
-            if feed_tip is None:
-                logger.warning("Unable to fetch feed tip")
-                return None
-
-            tip += feed_tip
+        feed_tip = await get_feed_tip(datafeed.query.query_id, self.autopay)
+        if feed_tip is None:
+            logger.warning("Unable to fetch feed tip")
         else:
-            suggested_qtag, autopay_tip = await autopay_suggested_report(self.autopay)
+            tip += feed_tip
 
-            if suggested_qtag is None:
-                suggested_qtag = await tellor_suggested_report(self.oracle)
+        return tip
 
+    async def fetch_datafeed(self) -> Optional[DataFeed[Any]]:
+        if self.datafeed:
+            self.autopaytip = await self.rewards()
+            return self.datafeed
+
+        suggested_qtag, autopay_tip = await autopay_suggested_report(self.autopay)
+        if suggested_qtag:
```

Comment:
Shouldn't it check if both returned values are not `None`, or?

Reply:
this checks if there is q tag suggestion if so autopay tip is at least zero and returns those values.  ```if suggested q_tag is None``` is checked and handled in subsequent lines